pkg/restore/restore_test.go

```diff
@@ -1378,7 +1408,7 @@ func TestRestoreActionAdditionalItems(t *testing.T) {
 			},
 		},
 		{
-			name:    "when using a restore namespace filter, additional items that are cluster-scoped are restored",
+			name:    "when using a restore namespace filter, additional items that are cluster-scoped are restored when IncludeClusterResources=nil",
```

Comment:
I know I'm probably missing something. In this use-case, the additional items are being restored (by default, since `IncludeClusterResources=nil`). Why am I thinking that they should not be restored in this case?

Reply:
`IncludeClusterResources` is a nullable bool, so has 3 possible modes:

- `true`: all cluster-scoped resources that are selected by the resource/label selectors are restored
- `false`: no cluster-scoped resources are restored
- `nil` ("auto"): only cluster-scoped resources that are dependencies of other namespaced resources being restored, are restored (e.g. we're restoring a pod that uses a PVC/PV, so we'd restore the PV as well). the way this is actually implemented in the code is by allowing "additional items" to be restored, even if they're cluster-scoped, when `IncludeClusterResources=nil`, since "additional items" are how we capture relationships between resources.

This logic already holds for backups, but we had a bug in the restore code that didn't implement the `nil` logic correctly.

Reply:
I appreciate the explanation. This also should be in the docs, tho I'm not sure where. I'm opening a tix.